src/util.js

```diff
@@ -406,19 +408,32 @@
    */
   connect.PopupManager = function () { };
 
-  connect.PopupManager.prototype.open = function(url, name) {
+  connect.PopupManager.prototype.open = function (url, name, options) {
     var then = this._getLastOpenedTimestamp(name);
     var now = new Date().getTime();
-    var win = null;      
+    var win = null;
     if (now - then > ONE_DAY_MILLIS) {
-       win = window.open('', name);
-       if (win.location !== url) {
+      if (options) {
+        // default values are chosen to provide a minimum height without scrolling
+        // and a uniform margin based on the css of the ccp login page
+        var height = options.height ? options.height : DEFAULT_POPUP_HEIGHT;
+        var width = options.width ? options.width : DEFAULT_POPUP_WIDTH;
+        var top = isNaN(options.top) ? 0 : options.top;
+        var left = isNaN(options.left) ? 0 : options.left;
+        win = window.open('', name, "width="+width+", height="+height+", top="+top+", left="+left);
+        if (win.location !== url) {
+          win = window.open(url, name, "width="+width+", height="+height+", top="+top+", left="+left);
+        }
+      } else {
+        win = window.open('', name);
+        if (win.location !== url) {
           win = window.open(url, name);
-       }
-       this._setLastOpenedTimestamp(name, now);
+        }
+      }
+      this._setLastOpenedTimestamp(name, now);
     }
     return win;
- };
+  };
```

Comment:
extra space here

Reply:
Before, this was indented one space less than `connect.PopupManager.prototype.open` above.